### PR TITLE
task1/jdbc: Class that implements basic queries in database

### PR DESCRIPTION
**Pull request by qnislavnachev**
There is a class that implements most basic queries which contains the
following commands: insert, update, delete, select, alter, drop, where,
like.

### CODE REVIEW

**Comment by qnislavnachev** [comment]
This pull request is not ready for review yet , because I have to made some changes.

**Comment by qnislavnachev** [comment]
@mgenov rewriten test now they are isolated

File: src/task1/Queries.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private String table;
+
+
+    public Queries(Connection connection, String table) throws SQLException {
+        this.table = table;
+        this.connection = connection;
+        statement = connection.createStatement();
+    }
+
+    public void printTable() throws SQLException {
```
Comment (mgenov) [inline src/task1/Queries.java]: printTable is common name but it uses concrete fields "Name", "Age", "Course" which is wrong. 

The name of the object should be common or concrete, but not both. 

The design should be:
```java
public interface EmployeeRepository {

  void register(Employee employee);

  List<Employee> findEmployees();
}
```

or 

```java
public interface Tables {
  // list all table rows
  List<Object[]> listTableRows(String tableName);

  // returns  updated rows
  int update(String tableName, String[] columns, Object[] values); 

}
```

The test should be updated depending on the used structure.

File: test/queriestest/TestQueries.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        }
+        assertThat(name, is("Iani"));
+        dropStudentsTable();
+    }
+
+    @Test
+    public void deleteInfo() throws Exception {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(out));
+        queries.insert(1, "Iani", 23, 3);
```
Comment (qnislavnachev) [inline test/queriestest/TestQueries.java]: It will be fixed

File: test/queriestest/TestQueries.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        String output = out.toString();
+        assertFalse(output.contains("Iani"));
+        System.setOut(System.out);
+        out.close();
+        dropStudentsTable();
+    }
+
+    @Test
+    public void addColumn() throws Exception {
+        queries.addColumn("Location", "varchar(20)");
```
Comment (qnislavnachev) [inline test/queriestest/TestQueries.java]: Null value

File: test/queriestest/TestQueries.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        while (resultSet.next()) {
+            studentLocation = resultSet.getString("Location");
+        }
+        assertThat(studentLocation, is(nullValue()));
+        dropStudentsTable();
+    }
+
+    @Test
+    public void dropTable() throws Exception {
+        queries.dropTable();
```
Comment (qnislavnachev) [inline test/queriestest/TestQueries.java]: that will be fixed

File: src/task1/Queries.java (hunk 1 above)
Comment (qnislavnachev) [inline src/task1/Queries.java]: I will fix that

**Comment by qnislavnachev** [comment]
last fixes done renamed methods now they are more understandable

File: src/task1/StudentsRepository.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
+package task1;
+
+import java.sql.*;
+import java.sql.Connection;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class StudentsRepository {
+    private Statement statement;
```
Comment (mgenov) [inline src/task1/StudentsRepository.java]: For each operation you need a statement and this class is having a few.

File: test/queriestest/TestQueries.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    private ResultSet createResultSet(String query) throws SQLException {
+        return connection.createStatement().executeQuery(query);
+    }
+
+    @Test
+    public void insertInfoToStudents() throws Exception {
+        Student iani = new Student(1, "Iani", 23, 2);
+        students.register(iani);
```
Comment (mgenov) [inline test/queriestest/TestQueries.java]:
```java
students.register(iani);
List<Student> allStudents = students.findAllStudents();
assertThat(allStudents.size(), is(1));
assertThat(allStudents.get(0).name, is(equalTo("Iani"));
```

**Comment by qnislavnachev** [comment]
I made some refactoring

**Comment by mgenov** [comment]
What was Refactored ? You will save me time If I know where to look.

**Comment by qnislavnachev** [comment]
renamed last commit because it was wrong named, this commit contain queries method changes, no longer they throws sqlexception and removed result set from most of tests

**Comment by qnislavnachev** [comment]
some little changes made now deleting and updating info by person ID

**Pull request merged**